test/framework/toy_build.py

```diff
@@ -198,19 +198,24 @@ def test_toy_build(self, extra_args=None, ec_file=None, tmpdir=None, verify=True
         # make sure full test report was dumped, and contains sensible information
         if test_report is not None:
             self.assertTrue(os.path.exists(test_report))
-            if fails:
-                test_result = 'FAIL'
+            if parse_error_regex:
```

Comment:
rename `parse_error_regex` to `test_report_regex`, and maybe make it a list (`test_report_regexs`) to allow passing in multiple patterns?

Reply:
b037800 and 4799348